File: client/Ciphelet/app/src/main/java/com/github/worldoneo/ciphelet/connector/api/LoginRequiredService.java
package com.github.worldoneo.ciphelet.connector.api;

import com.github.worldoneo.ciphelet.connector.CipheletAPI;
import com.github.worldoneo.ciphelet.connector.Connector;
import com.github.worldoneo.ciphelet.connector.action.GenericAction;
import com.github.worldoneo.ciphelet.connector.events.EventManager;
import com.github.worldoneo.ciphelet.connector.events.LoginRequiredEvent;

public class LoginRequiredService extends Service {
    public LoginRequiredService(Connector connector, CipheletAPI cipheletAPI) {
        super(connector, cipheletAPI, GenericAction.Action.LOGIN_REQUIRED);
    }

    @Override
    public void receivedAction(GenericAction action) {
        EventManager.getInstance().handleEvent(new LoginRequiredEvent());
    }
}

File: client/Ciphelet/app/src/main/java/com/github/worldoneo/ciphelet/connector/api/LoginRequiredService.kt
package com.github.worldoneo.ciphelet.connector.api

import com.github.worldoneo.ciphelet.connector.CipheletAPI
import com.github.worldoneo.ciphelet.connector.Connector
import com.github.worldoneo.ciphelet.connector.action.GenericAction
import com.github.worldoneo.ciphelet.connector.events.EventManager.handleEvent
import com.github.worldoneo.ciphelet.connector.events.LoginRequiredEvent

class LoginRequiredService(connector: Connector?, cipheletAPI: CipheletAPI) : Service(connector, cipheletAPI, GenericAction.Action.LOGIN_REQUIRED) {
    override fun receivedAction(action: GenericAction) {
        handleEvent(LoginRequiredEvent())
    }
}